### DISCUSSION TITLE
GraphQL: Enterprise cloud user query - Hint needed

### DISCUSSION THREAD

**Discussion by Timmmy-nlb** [community/community#49491] [Enterprise] [answered]
### Select Topic Area

Question

### Body

I am currently trying my hand at graph queries with enterprise cloud.

Retrieving the organizations of enterprise cloud is easy...
```
{enterprise(slug: "myEnterprise") {
    organizations(first: 10) {
      edges {
        node {
          name
        }
      }
    }
  }
}
```

But how to retrieve the members? (without SAML provider)
I always get only NUL as return values.

(Or how to check if a specific user is enterprise member).
```
{enterprise(slug: "myEnterprise") {
    members(first: 10) {
      edges {
        node {
          ... on User {
            id
            name
            login
          }
        }
      }
    }
  }
}
```

```
{"data": {
    "enterprise": {
      "members": {
        "edges": [
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null
        ]
      }
    }
  }
}
```
Anybody have a hint?

**Comment by Timmmy-nlb** [comment]
Hello @SayedAlAhlMohamed,
i have query the statement directly and get a error, that 'enterprise' is not in context at user (visible in the graphical explorer of github): 

```
{
  "errors": [
    {
      "path": [
        "query",
        "user",
        "enterprise"
      ],
      "extensions": {
        "code": "undefinedField",
        "typeName": "User",
        "fieldName": "enterprise"
      },
      "locations": [
        {
          "line": 3,
          "column": 5
        }
      ],
      "message": "Field 'enterprise' doesn't exist on type 'User'"
    }
  ]
} 
```

**Comment by Timmmy-nlb** [comment]
Sorry @SayedAlAhlMohamed,
but this did'nt work too:
Because below the node is no value "id"...

![grafik](https://user-images.githubusercontent.com/52195132/224036048-78a2e7df-5fb0-454b-bd21-0f412ef5018a.png)
![grafik](https://user-images.githubusercontent.com/52195132/224035603-413817cb-ed46-493b-b6b9-db16ef9cba89.png)

**Comment by Timmmy-nlb** [comment]
Anyone else a hint?

**Comment by DrNeil** [comment, marked as answer]
found this issue today and how to get it working 
firstly you need to have the correct authention scopes 
I am using GH CLI and this gets the correct tokens 
`gh auth login -p https -h GitHub.com -s repo,user,admin:enterprise,admin:org -w`

Then the query looks like this 
```
$query = @'
query($cursor: String) {
  enterprise(slug: "<YOUR ENTERPRISE SLUG") {
    members(first: 100, after: $cursor, role:MEMBER) {
      pageInfo {
        hasNextPage
        endCursor
      }
      nodes{ 
        ... on EnterpriseUserAccount{ 
          login
          id, 
          organizations(first:2){ 
            nodes{
              ... on Organization{
                login
              }
            }
          }
        }
      } 
    }
  }
}
'@

```

This gets the first two Enterprise organizations for each Enterprise Member 

Then to get the result 
`$response = gh api graphql -f query="$query" -f cursor="$cursor" `
